http-server/src/main/java/io/strimzi/http/server/AdminServer.java

```diff
@@ -0,0 +1,53 @@
+package io.strimzi.http.server;
+
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.http.HttpServer;
+import io.vertx.ext.web.Router;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.ServiceLoader;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.javatuples.Triplet;
+
+public class AdminServer extends AbstractVerticle {
+    private static final Logger LOGGER = LogManager.getLogger(AdminServer.class);
+
+    @Override
+    public void start(final Promise<Void> startServer) {
+
+        loadRoutes()
+            .onSuccess(routes -> {
+                final HttpServer server = vertx.createHttpServer();
+                final Router router = Router.router(vertx);
+                router.mountSubRouter("/", routes);   // Mount the sub router containing the module routes
+                server.requestHandler(router).listen(8080);
+                LOGGER.info("AdminServer is listening on port 8080");
+            })
+            .onFailure(throwable -> LOGGER.atFatal().withThrowable(throwable).log("Loading of routes was unsuccessful."));
+    }
+
+    private Future<Router> loadRoutes() {
+
+        final Router router = Router.router(vertx);
```

Comment:
you are creating a `Router` here, mounting subrouters returned by the registered services but then on line 25, you create another `Router` adding a subrouter which is the previous router with related subrouters. Can we avoid the `Router` in the middle?